plugins/http_plugin/include/eosio/http_plugin/http_plugin.hpp

```diff
@@ -88,25 +94,24 @@ namespace eosio {
         void plugin_shutdown();
         void handle_sighup() override;
 
-        void add_handler(const string& url, const url_handler&, appbase::exec_queue q, int priority = appbase::priority::medium_low, http_content_type content_type = http_content_type::json);
-        void add_api(const api_description& api, appbase::exec_queue q, int priority = appbase::priority::medium_low, http_content_type content_type = http_content_type::json) {
-           for (const auto& call : api)
-              add_handler(call.first, call.second, q, priority, content_type);
+        void add_handler(api_entry&& entry, appbase::exec_queue q, int priority = appbase::priority::medium_low, http_content_type content_type = http_content_type::json);
+        void add_api(api_description&& api, appbase::exec_queue q, int priority = appbase::priority::medium_low, http_content_type content_type = http_content_type::json) {
+           for (auto& call : api)
+              add_handler(std::move(call), q, priority, content_type);
         }
 
-        void add_async_handler(const string& url, const url_handler& handler, http_content_type content_type = http_content_type::json);
-        void add_async_api(const api_description& api, http_content_type content_type = http_content_type::json) {
-           for (const auto& call : api)
-              add_async_handler(call.first, call.second, content_type);
+        void add_async_handler(api_entry&& entry, http_content_type content_type = http_content_type::json);
+        void add_async_api(api_description&& api, http_content_type content_type = http_content_type::json) {
+           for (auto& call : api)
+              add_async_handler(std::move(call), content_type);
         }
 
         // standard exception handling for api handlers
         static void handle_exception( const char *api_name, const char *call_name, const string& body, const url_response_callback& cb );
 
         void post_http_thread_pool(std::function<void()> f);
 
-        bool is_on_loopback() const;
-        bool is_secure() const;
+        bool is_on_loopback(api_category category) const;
 
         static bool verbose_errors();
 
```

Comment:
`get_supported_apis_result get_supported_apis()const;` below is not implemented.